perfkitbenchmarker/publisher.py

```diff
@@ -487,8 +487,11 @@ class SampleCollector(object):
       SampleCollector._DefaultPublishers.
     run_uri: A unique tag for the run.
   """
-  def __init__(self, metadata_providers=None, publishers=None):
-    self.samples = []
+  def __init__(self, samples=None, metadata_providers=None, publishers=None):
+    if samples is None:
```

Comment:
Changed this to
```
self.samples = [] if samples is None else samples
```
reason being that passing in a Falsey value like an empty list shouldn't create a new list.